blockchain-agent-proof/src/main/java/com/sphereon/alfresco/blockchain/agent/proof/config/BlockchainProofApiConfig.java

```diff
@@ -19,22 +20,30 @@
 import java.util.concurrent.TimeUnit;
 
 @Configuration
-public class BlockchainApiConfig {
-    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(BlockchainApiConfig.class);
-
-    @Value("${sphereon.blockchain-proof-api.base-path:https://gw.api.cloud.sphereon.com/blockchain/proof/0.10}")
-    String bcProofBasePath;
-
-    @Value("${sphereon.blockchain-proof-api.base-path:https://gw.api.cloud.sphereon.com/crypto/keys/0.9}")
-    String cryptoBasePath;
+public class BlockchainProofApiConfig {
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(BlockchainProofApiConfig.class);
+
+    private String bcProofBasePath;
+    private int connectionTimeout;
+    private String applicationName;
+
+    public BlockchainProofApiConfig(@Value("${sphereon.blockchain.agent.blockchain-proof.application-name") final String applicationName,
+                                    @Value("${sphereon.blockchain.agent.blockchain-proof.base-path:https://gw.api.cloud.sphereon.com/blockchain/proof/0.10}") final String bcProofBasePath,
+                                    @Value("${sphereon.blockchain.agent.blockchain-proof.api-client.timeout:120000}") final int connectionTimeout) {
+        this.applicationName = applicationName;
+        this.bcProofBasePath = bcProofBasePath;
+        this.connectionTimeout = connectionTimeout;
+    }
 
-    @Value("${sphereon.api-client.timeout:120000}")
-    int connectionTimeout;
+    @Bean
+    public RequestHeaderAccess requestHeaderAccess() {
+        return new RequestHeaderAccess();
+    }
 
     @Bean
     AuthenticationApi authenticationApi() {
-        ApiConfiguration.Builder configBuilder = new ApiConfiguration.Builder()
-                .withApplication("alfresco-blockchain")
+        final var configBuilder = new ApiConfiguration.Builder()
+                .withApplication(applicationName)
```

Comment:
add TODO, to move this from envVars to application properties. This is why the Sphereon Blockchain Proof has to be partly configured using env variables right now in the application.properties

Reply:
Yes, I'm happy with that suggestion. Added the TODO.